File: Java/LeetCode/Difference Between Element Sum and Digit Sum of an Array.java
// https://leetcode.com/problems/difference-between-element-sum-and-digit-sum-of-an-array/

class Solution {

    public int differenceOfSum(int[] nums) {
        int elementSum = 0;
        int digitSum = 0;

        for (int num : nums) {
            elementSum += num;
            String numStr = String.valueOf(num);
            for (int i = 0; i < numStr.length(); ++i) {
                int digit = Character.getNumericValue(numStr.charAt(i));
                digitSum += digit;
            }
        }

        return Math.abs(elementSum - digitSum);
    }
    
}

File: Kotlin/LeetCode/Difference Between Element Sum and Digit Sum of an Array.kt
// https://leetcode.com/problems/difference-between-element-sum-and-digit-sum-of-an-array

class Solution {

    fun differenceOfSum(nums: IntArray): Int {
        var elementSum = 0
        var digitSum = 0

        nums.forEach {
            elementSum += it;
            val numStr = it.toString()
            for (c in numStr) {
                digitSum += Character.getNumericValue(c);
            }
        }

        return Math.abs(elementSum - digitSum);
    }

}
